File: js/dates.js
function title(v)
{
  console.log("-");
  console.log(v);
}

function observation(v)
{
  console.log(v);
}

function var_export(v)
{
  console.log(v);
}



title('today as ISO');
var res1 = (new Date()).toISOString().substr(0, 10);
var_export(res1);

title('yesterday as ISO');
var res2 = new Date();
res2.setDate(res2.getDate() - 1);
res2 = res2.toISOString().substr(0, 10);
var_export(res2);

title('last day of this month as ISO');
var res3 = (new Date((new Date()).getFullYear(), (new Date()).getMonth() + 1, 0)).getDate();
var_export(res3);

title('last day of given month as ISO');
var res4 = (new Date(2014, 2 - 1 + 1, 0)).getDate();
var_export(res4);

File: perl/dates.pl
use strict;
use warnings;

use Time::Piece;
use Time::Seconds;
use Data::Dumper;

sub title
{
  my $v = shift;
  print "\n\n\n\n$v\n\n";
}

sub var_dump
{
  my $v = shift;
  print Dumper $v;
}

sub observation
{
  my $v = shift;
  print "\n$v";
}



print "Content-Type: text/html\n\n";
print "<pre>";



title('today as ISO');
my $res1 = localtime->ymd;
var_dump($res1);

title('yesterday as ISO');
my $res2 = (Time::Piece->new - ONE_DAY)->ymd;
var_dump($res2);

title('last day of this month as ISO');
my $res3 = localtime->month_last_day;
var_dump($res3);

title('last day of given month as ISO');
my $res4 = Time::Piece->strptime("2014-02", "%Y-%m")->month_last_day;
var_dump($res4);



print "</pre>";

